dsn.go

```diff
@@ -390,16 +390,11 @@ func fillMissingConfigParameters(cfg *Config) error {
 		return ErrEmptyAccount
 	}
 
-	if cfg.Authenticator != AuthTypeOAuth && strings.Trim(cfg.User, " ") == "" {
-		// oauth does not require a username
+	if authRequiresUser(cfg) && strings.Trim(cfg.User, " ") == "" {
```

Comment:
`authRequiredUsername`?

Reply:
I would stick to the "domain" name of this property, which is "user" in this project.

Reply:
Alright. A bit misleading though.